python/graphstorm/model/rgcn_encoder.py

```diff
@@ -504,3 +589,112 @@ def forward(self, g, inputs, mod_args=None, mod_kwargs=None):
             if len(alist) != 0:
                 rsts[nty] = self.agg_fn(alist, nty)
         return rsts
+
+
+class GraphConvwithEdgeFeat(nn.Module):
+    """ Graph conv module with edge feature supported in message passing computation.
+
+    .. versionadded:: 0.4.0
+        Add `GraphConvwithEdgeFeat` class in v0.4.0 to support edge feature in message
+        passing computation.
+
+    Parameters
+    ----------
+    in_feat: int
+        Input feature size.
+    out_feat: int
+        Output feature size.
+    edge_feat_mp_op: str
+        The opration method to combine source node embeddings with edge embeddings in message
+        passing. Options include `concat`, `add`, `sub`, `mul`, and `div`.
+    bias: bool
+        Whether to add bias. Default: True.
+    """
+    def __init__(
+        self,
+        in_feat,
+        out_feat,
+        edge_feat_mp_op='concat',
+        bias=True):
+        super(GraphConvwithEdgeFeat, self).__init__()
+        self.in_feat = in_feat
+        self.out_feat = out_feat
+        assert edge_feat_mp_op in BUILTIN_EDGE_FEAT_MP_OPS, 'GraphStorm only support edge' + \
+            f' message passing operation in {BUILTIN_EDGE_FEAT_MP_OPS}, bug got ' + \
+            f'{edge_feat_mp_op}.'
+        self.edge_feat_mp_op = edge_feat_mp_op
+        self.bias = bias
+
```

Comment:
We assume in_feat and out_feat are the same here. If you take a look at https://docs.dgl.ai/en/2.0.x/_modules/dgl/nn/pytorch/conv/graphconv.html#GraphConv, dgl.GraphConv handle the case when in_feat is different from out_feat.

Reply:
The current implementation support different input and output dimension. But do require input node feature and input edge feature to be the same to ensure their operations.